diff --git a/node_modules/react-native-worklets-core/android/build.gradle b/node_modules/react-native-worklets-core/android/build.gradle
index b6af1c3..3d1bba3 100644
--- a/node_modules/react-native-worklets-core/android/build.gradle
+++ b/node_modules/react-native-worklets-core/android/build.gradle
@@ -82,7 +82,8 @@ android {
                 "-DANDROID_TOOLCHAIN=clang",
                 "-DREACT_NATIVE_DIR=${nodeModules}/react-native",
                 "-DJS_RUNTIME=${JS_RUNTIME}",
-                "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
+                "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
+                "-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-z,max-page-size=16384"
         abiFilters (*reactNativeArchitectures())
       }
     }
